onnxruntime/core/providers/qnn/builder/qnn_model.cc

```diff
@@ -210,13 +236,19 @@ Status QnnModel::ExecuteGraph(const Ort::KernelContext& context, const logging::
     auto ort_input_tensor = context.GetInput(qnn_input_info.ort_index);
     auto ort_tensor_size = TensorDataSize(ort_input_tensor);
     LOGS(logger, VERBOSE) << "Qnn tensor size: " << qnn_input_info.tensor_byte_size
-                          << "Ort tensor size: " << ort_tensor_size;
+                          << " Ort tensor size: " << ort_tensor_size;
     ORT_RETURN_IF_NOT(qnn_input_info.tensor_byte_size == ort_tensor_size,
                       "ORT Tensor data size does not match QNN tensor data size.");
 
     qnn_inputs.push_back(qnn_input_info.tensor_wrapper->GetQnnTensor());
-    SetQnnTensorClientBuf(qnn_inputs.back(),
-                          const_cast<void*>(ort_input_tensor.GetTensorData<void>()), qnn_input_info.tensor_byte_size);
+
+    ORT_RETURN_IF_ERROR(BindQnnTensorMemoryToOrtValue(
+        logger,
+        *qnn_backend_manager_,
+        *static_cast<const OrtMemoryInfo*>(ort_input_tensor.GetTensorMemoryInfo()),
+        const_cast<void*>(ort_input_tensor.GetTensorRawData()), qnn_input_info.tensor_byte_size,
```

Comment:
nit: can we do the const_cast in BindQnnTensorMemoryToOrtValue so it's in less places?

Reply:
got it down to one place. I think it's better to leave const_cast at the call site since this function actually expects a non-const void*.

Reply:
Shall we add a comment for this?